### ISSUE TITLE
NaturalProofs (Eval Task)

### ISSUE THREAD

**Issue by haileyschoelkopf** [EleutherAI/math-lm#4]

**Comment by zhangir-azerbayev** [comment]
@wellecks I suspect we should be worried about data contamination. How did you deal with this in the original naturalprover paper? Will that same methodology still work in this context?

**Comment by wellecks** [comment]
Good points. Here are some thoughts:

1. **Naturalprover:** In the naturalprover paper, we mitigated contamination _from the finetuning dataset_ by using the NaturalProofs splits, which ensure that validation or test proofs are not seen in the finetuning dataset. However, _we did not address contamination from the **pretraining** corpus_. For the gpt-3 based models we did not have access to the pretraining corpus, and for the gpt-j-6b model we didn't look whether proofwiki is in the Pile.

2. **LM-evaluation-harness:** Adding it to the lm-evaluation-harness would give us a way to press a button and get informal proofs from a well-defined domain (the domain of ProofWiki theorems). I think this makes it still interesting to add from a scientific perspective, perhaps with disclaimers about the potential for contamination. For instance, we could even _use it to study contamination_; are certain generated proofs actually attributed to contamination (which requires developing relevant notions of contamination)? 

3. **Temporal split:** we could potentially create a new split based on theorems that were added past a certain date. This information should be available in the xml dump of proofwiki. However, this would be its own small side project.

4. **Our model:** we could try to only include the naturalprover training split in the corpus (versus the splits used in the original proofpile), and e.g. exclude proofwiki from the web scrape (naturally, this doesn't prevent all potential for contamination).
